test/integration/simulacron.js

```diff
@@ -0,0 +1,294 @@
+'use strict';
+var helper = require('../test-helper');
+var http = require('http');
+var spawn = require('child_process').spawn;
+var util = require('util');
+var fs = require('fs');
+
+var simulacronHelper = {
+  _execute: function(processName, params, cb) {
+    var originalProcessName = processName;
+
+    // If process hasn't completed in 10 seconds.
+    var timeout = undefined;
+    if(cb) {
+      timeout = setTimeout(function() {
+        cb("Timed out while waiting for " + processName + " to complete.");
+      }, 10000);
+    }
+
+    var p = spawn(processName, params, {});
+    p.stdout.setEncoding('utf8');
+    p.stderr.setEncoding('utf8');
+    p.stdout.on('data', function (data) {
+      helper.trace("%s_out> %s", originalProcessName, data);
+    });
+
+    p.stderr.on('data', function (data) {
+      helper.trace("%s_err> %s", originalProcessName, data);
+    });
+
+    p.on('close', function (code) {
+      helper.trace("%s exited with code %d", originalProcessName, code);
+      if(cb) {
+        clearTimeout(timeout);
+        if (code === 0) {
+          cb();
+        } else {
+          cb(Error("Process exited with non-zero exit code: " + code));
+        }
+      }
+    });
+
+    return p;
+  },
+  start: function(cb) {
+    var self = this;
+    var simulacronJarPath = process.env['SIMULACRON_PATH'];
+    if (!simulacronJarPath) {
+      simulacronJarPath = "$HOME/simulacron.jar";
+      helper.trace("SIMULACRON_PATH not set, using $home/simulacron.jar");
+    }
+    if (!fs.existsSync(simulacronJarPath)) {
+      throw new Error("Simulacron jar not found at: " + simulacronJarPath);
+    }
+
+    var processName = 'java';
+    var params = ['-jar', simulacronJarPath, '--ip', '127.0.0.101'];
+    var initialized = false;
+
+    var timeout = setTimeout(function() {
```

Comment:
I think this `timeout` is not necessary, as you already handled it in `_execute()` method.

Reply:
They are different timeouts. The one inside _execute just check if the process timeouts to start. The one inside start verify if the process start correctly (through logs).